controllers/defaults/placements.go

```diff
@@ -50,26 +51,12 @@ var (
 			},
 			PodAntiAffinity: &corev1.PodAntiAffinity{
 				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
-					getWeightedPodAffinityTerm(100, "rook-ceph-osd-prepare"),
+					getWeightedPodAffinityTerm(100, osdLabelSelector),
+					getWeightedPodAffinityTerm(100, osdPrepareLabelSelector),
 				},
 			},
-		},
-
-		"osd-tsc": {
-			Tolerations: []corev1.Toleration{
-				getOcsToleration(),
-			},
 			TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
-				getTopologySpreadConstraintsSpec(1),
-			},
-		},
-
-		"osd-prepare-tsc": {
-			Tolerations: []corev1.Toleration{
-				getOcsToleration(),
-			},
-			TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
-				getTopologySpreadConstraintsSpec(1),
+				getTopologySpreadConstraintsSpec(1, osdPrepareLabelSelector),
```

Comment:
Did you intend to add the `osdLabelSelector` as well on this TSC for the prepare pods? Looks like it was only added on the pod anti-affinity above.

Reply:
I see upstream in the TSC we have only the osd prepare label in the tsc too, so I kept like that, Do we need it differently here?

Reply:
Oh right, the upstream example would also have the same issue. I guess it's not commonly hit since osd prepare jobs are not commonly deleted manually. 

Still we need to have the TSC for the OSD prepare as suggested in [this comment](https://github.com/red-hat-storage/ocs-operator/pull/2448#discussion_r1481882583).

Reply:
Yeah updated now.